projects/exui-common-lib/src/lib/services/idle/idle.services.ts

```diff
@@ -0,0 +1,68 @@
+import { Injectable } from '@angular/core';
+import {DocumentInterruptSource, Idle, } from '@ng-idle/core';
+import {Keepalive} from '@ng-idle/keepalive';
+import {Observable, Subject} from 'rxjs';
+import {
+  delay,
+  distinctUntilChanged,
+  map
+} from 'rxjs/operators';
+import {IdleConfigModel} from '../../models/idle-config.model';
+
+@Injectable()
+export class IdleService {
+  private appStateEmitter: Subject<{type: string, countdown?: string; isVisible?: boolean}>;
```

Comment:
Try not to use inline typings for complex types, but rather make it a proper model that's exported so that implementing apps can use those types for their own type-checking as well.